cpp/src/table/primitive_row_operators.cu

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/table/primitive_row_operators.cuh>
+#include <cudf/types.hpp>
+#include <cudf/utilities/traits.hpp>
+
+#include <algorithm>
+
+namespace cudf {
+bool is_primitive_row_op_compatible(cudf::table_view const& table)
+{
+  return table.num_columns() == 1 and std::all_of(table.begin(), table.end(), [](auto const& col) {
+           return cudf::is_numeric(col.type());
+         });
+}
```

Comment:
Is this small function worth to be separated into a separate source file?

Reply:
This small function is used across multiple translation units. To help reduce build time and binary size, it's generally better to separate the definition from the declaration whenever possible.

Reply:
Yes, but I meant this function is too small thus I'm not sure if doing this would be any helpful.

Reply:
It's not just about build time and binary size. Separating declarations and definitions is generally considered good practice in C++, even for trivial implementations.

Reply:
Yes, I think this makes sense to separate. Thanks!